src/main/java/net/dv8tion/jda/core/entities/Invite.java

```diff
@@ -0,0 +1,275 @@
+package net.dv8tion.jda.core.entities;
+
+import net.dv8tion.jda.core.JDA;
+import net.dv8tion.jda.core.entities.impl.InviteImpl;
+import net.dv8tion.jda.core.requests.RestAction;
+
+import java.time.OffsetDateTime;
+
+/**
+ * Representation of a Discord Invite.
+ * This class is immutable.
+ *
+ * @since  JDA 3.0
+ * @author Aljoscha Grebe
+ */
+public interface Invite
+{
+    /**
+     * Returns a new {@link Invite} for the given invite code.
+     *
+     * <br>Possible {@link net.dv8tion.jda.core.requests.ErrorResponse ErrorResponses}:
+     * <ul>
+     * <li>{@link net.dv8tion.jda.core.requests.ErrorResponse#UNKNOWN_INVITE Unknown Invite} <br>
+     * The Invite did not exist (possibly deleted)</li>
+     * </ul>
+     *
+     * @param api
+     *        The JDA object
+     * @param code
+     *        The invite code
+     *
+     * @throws net.dv8tion.jda.core.exceptions.ErrorResponseException
+     *
+     * @return {@link net.dv8tion.jda.core.requests.RestAction RestAction} -
+     *
+     *         <pre>
+     * Type: {@link net.dv8tion.jda.core.entities.Invite Invite}
+     *         <br>Value: {@code The Invite object}
+     *         </pre>
+     */
+    public static RestAction<Invite> resolve(final JDA api, final String code)
```

Comment:
Minn is right here. `public` can (and should) always be omitted from interfaces.